### DISCUSSION TITLE
v0.23.0: onRecordsListRequest empty record

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#5670] [Q&A] [answered]
I want to modify the start_date record, but the record is empty for some reason.. 
but when I look in item with console.log start_date is filled. How can I modify the record?

```
onRecordsListRequest((e) => {

    const auth = e.auth
    const adminAuth = e.hasSuperuserAuth()

    if (!auth || adminAuth) return;

    e.result.items.forEach(item => {
        console.log(item.start_date)
    })

    e.next()
}, "workbooks")
````

**Comment by ganigeorgiev** [comment]
At the moment record fields can be accessed or modified only via the designated `get`/`set` methods, aka. try with `item.get("start_date")`.

**Reply by ganigeorgiev** [reply]
_Side-note: the `if` condition return should return `e.next()`._

**Reply by starredev** [reply]
> _Side-note: the `if` condition return should return `e.next()`._

Alright, that works, thanks!

If I want to add a record, I could add ```record.set("date", newDate)``` in the for loop?
Because I tried, but it seems it didnt add it

For example
````
onRecordsListRequest((e) => {

    const auth = e.auth
    const adminAuth = e.hasSuperuserAuth()

    if (!auth || adminAuth) return e.next()
    const records = e.result.items;

    records.forEach(record => {
        // Get the 'start_date' field
        let startDate = new Date(record.get("start_date"));

        // Modify the date, e.g., add 1 day
        startDate.setDate(startDate.getDate() + 1);

        // Update the 'start_date' back into the record
        record.set("new_date", startDate.toISOString());
    });

    e.next()
});
````

**Reply by ganigeorgiev** [reply]
I'm tired and don't have the energy to go through another back-and-forth. The above doesn't make sense to me. Please elaborate clearly what you are trying to do and provide some context about your available collection fields.

If you want to adjust some record field value only for the serialization, consider using the `onRecordEnrich` hook (it is called for list, view, create, update, realtime events, etc.).

**Reply by starredev** [reply]
Sorry that was not my intention.

I want that when someone requests a list of the workbooks collection, an existing record is modified and added as a new record to the object. I didn't know about onRecordEnrich. Will look into it. Thanks

**Reply by ganigeorgiev** [reply, marked as answer]
I suspect that you are confusing "record" with "field". If in the above "new_data" is not part of the collection fields definition, then you need to explicitly allow custom data serialization in order to be returned in the json:
```js
onRecordEnrich((e) => {
    ...
    e.record.withCustomData(true)
    e.record.set("someCustomField", ...)
    
    e.next()
}, "yourCollection")
```
I'll stop engaging, but I doubt that your initial code previously worked with an older PocketBase version.

**Reply by starredev** [reply]
> ```js
> e.record.set("someCustomField", ...)
> ```

onRecordEnrich did the trick. Sorry for misunderstanding about "record" instead of field.  I never mentoined it worked earlier, I recently moved to 0.23 and was looking into documentation but couldn't really find what I needed. Also not explaining it well to you.. and because we don't understand each other I'm wasting your time. I feel sorry about that.

It worked, I'm happy thanks for you help @ganigeorgiev
